### PR TITLE
[API-27988] - Add GHA workflow for monitoring security alerts

### PR DESCRIPTION
**Pull request by jw81**
### Description

Original Issue :: [API-27988](https://jira.devops.va.gov/browse/API-27988)

- Adds a GitHub Action workflow that checks for any security vulnerability alerts in the repo and sends a Slack notification to the `#team-okapi-alerts` channel if any are found.

### CODE REVIEW

File: .github/workflows/notify-security-alerts.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          cat code-scanning.json
+          if [ "[]" == "$(cat code-scanning.json)" ]; then
+            echo "security_alert_exists='false'" >> $GITHUB_OUTPUT
+          else
+            echo "security_alert_exists='true'" >> $GITHUB_OUTPUT
+          fi
+      - id: check_for_dependabot_alerts
+        name: Check for dependabot alerts   
+        run: |
+          echo ${{ secrets.DEVPORTAL_SECURITY_SCAN_WORKFLOW_TOKEN }} | gh auth login --with-token
```
Comment (gileswells) [inline .github/workflows/notify-security-alerts.yml:26]: Shouldn't need to be repeated

File: .github/workflows/notify-security-alerts.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          cat dependabot.json
+          if [ "[]" == "$(cat dependabot.json)" ]; then
+            echo "security_alert_exists='false'" >> $GITHUB_OUTPUT
+          else
+            echo "security_alert_exists='true'" >> $GITHUB_OUTPUT
+          fi
+      - id: check_for_secret_scanning_alerts
+        name: Check for secret scanning alerts   
+        run: |
+          echo ${{ secrets.DEVPORTAL_SECURITY_SCAN_WORKFLOW_TOKEN }} | gh auth login --with-token
```
Comment (gileswells) [inline .github/workflows/notify-security-alerts.yml:38]: Shouldn't need to be repeated

**Review by gileswells — CHANGES_REQUESTED** [review]
Overall looks good, just a couple lines that can be removed to tidy it up.

File: .github/workflows/notify-security-alerts.yml (hunk 1 above)
Comment (jw81) [reply .github/workflows/notify-security-alerts.yml]: Ah, ok.
For some reason, I was thinking that `step` order wasn't guaranteed.  So I was thinking I needed to give each `step` some logic to make sure it was logged in.  But, it looks like that's [not the case](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#jobs).
> Steps are executed in order and are dependent on each other.

Removed in [this commit](https://github.com/department-of-veterans-affairs/developer-portal/pull/1311/commits/86e44fba51d89964ee78d47b52d808886c4cc390).

File: .github/workflows/notify-security-alerts.yml (hunk 2 above)
Comment (jw81) [reply .github/workflows/notify-security-alerts.yml]: Ah, ok.
For some reason, I was thinking that `step` order wasn't guaranteed.  So I was thinking I needed to give each `step` some logic to make sure it was logged in.  But, it looks like that's [not the case](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#jobs).
> Steps are executed in order and are dependent on each other.

Removed in [this commit](https://github.com/department-of-veterans-affairs/developer-portal/pull/1311/commits/86e44fba51d89964ee78d47b52d808886c4cc390).

**Review by gileswells — APPROVED** [review]

**Pull request merged**